integration-tests/tests/circuits.rs

```diff
@@ -60,32 +199,30 @@ async fn test_state_circuit_block(block_num: u64) {
     let storage_ops = builder.block.container.sorted_storage();
     trace!("storage_ops: {:#?}", storage_ops);
 
-    const DEGREE: usize = 17;
-
-    let rw_map = RwMap::from(&builder.block.container);
+    // TODO: use rw_map from block
+    // let rw_map = RwMap::from(&builder.block.container);
+    let rw_map = RwMap::default();
```

Comment:
We should fix this.  A `RwMap::default()` will just generate an empty state circuit, which is not verifying anything.
Did you get a failure when running the mock prover for this test?  Or was it for the real prover?

Reply:
```
thread 'serial_test_state_circuit_erc20_openzeppelin_transfer_succeed' panicked at 'proof generation should not fail: Synthesis', integration-tests/tests/circuits.rs:96:10
stack backtrace:
   0: _rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: circuits::test_run
   4: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
   6: tokio::runtime::scheduler::current_thread::Context::enter
   7: tokio::macros::scoped_tls::ScopedKey<T>::set
   8: tokio::runtime::scheduler::current_thread::CurrentThread::block_on
   9: tokio::runtime::Runtime::block_on
  10: core::ops::function::FnOnce::call_once
```

how do I find out where is the problem?

Reply:
When implementing the `Circuit` trait (from halo2), the error type for the `synthesize` method is already defined, and there's no way to add extra info in it.  The synthesis step usually does a lot of processing with the inputs (in order to calculate intermediate witness values), and there are many places that can return error.  Since we can't encode more details in the error itself, we have added lots of log::error right before the error is returned, like this:
https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/de26a9d4a8bfa1f416d65db805c091c4156d3ae9/zkevm-circuits/src/bytecode_circuit/bytecode_unroller.rs#L425-L430

Did you see any log before the panic backtrace?

Reply:
Solved by using new_from_block